pydrive2/fs/spec.py

```diff
@@ -542,6 +542,34 @@ def cp_file(self, lpath, rpath, **kwargs):
             buffer = io.BytesIO(stream.read())
             self.upload_fobj(buffer, rpath)
 
+    def mv(self, path1, path2, recursive=False, maxdepth=None, **kwargs):
+
+        if recursive:
+            raise NotImplementedError("Recursive move is not supported")
+        if maxdepth:
+            raise NotImplementedError("Max depth move is not supported")
+
+        file1_name = posixpath.basename(path1)
+        file2_name = posixpath.basename(path2)
+
+        file1_parent = self._parent(path1)
+        file2_parent = self._parent(path2)
```

Comment:
check: can any of those be None?

Reply:
if `file1_parent` is None, `file1_parent != file2_parent` is true and the function would try to move the file.

if `file2_parent` is None, `file2_parent_id = self._get_item_id(file2_parent)` will fail:

```python
Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/tests/gdrivefs.py", line 87, in <module>
    fs.mv('root/tmp/test/b.pdf', 'root/tmp/a.pdf')
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/funcy/decorators.py", line 45, in wrapper
    return deco(call, *dargs, **dkwargs)
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/funcy/flow.py", line 127, in retry
    return call()
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/funcy/decorators.py", line 66, in __call__
    return self._func(*self._args, **self._kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/fs/spec.py", line 569, in mv
    file2_parent_id = self._get_item_id(file2_parent)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/fs/spec.py", line 367, in _get_item_id
    bucket, base = self.split_path(path)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/fs/spec.py", line 239, in split_path
    parts = path.replace("//", "/").rstrip("/").split("/", 1)
AttributeError: 'NoneType' object has no attribute 'replace'
```

if `file2_parent` is `""`, `file2_parent_id = self._get_item_id(file2_parent)` will fail:

```python
Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/tests/gdrivefs.py", line 87, in <module>
    fs.mv('root/tmp/test/b.pdf', 'root/tmp/a.pdf')
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/funcy/decorators.py", line 45, in wrapper
    return deco(call, *dargs, **dkwargs)
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/funcy/flow.py", line 127, in retry
    return call()
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/funcy/decorators.py", line 66, in __call__
    return self._func(*self._args, **self._kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/fs/spec.py", line 569, in mv
    file2_parent_id = self._get_item_id(file2_parent)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/fs/spec.py", line 368, in _get_item_id
    assert bucket == self.root
AssertionError
```

Reply:
Q: do we need to handle this? I mean can the be actually None or empty?

Reply:
I don't think so, if it doesn't find the parent it will just throw a `FileNotFoundError: [Errno 2] No such file or directory:`